src/workspace-data/data-table/wds/AttributeInput.ts

```diff
@@ -0,0 +1,122 @@
+import _ from 'lodash/fp';
+import { Fragment, useEffect, useRef } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import * as Utils from 'src/libs/utils';
+
+import { renderInputForAttributeType } from '../shared/AttributeInput';
+import { getAttributeType } from './attribute-utils';
+
+const defaultValueForAttributeType = (attributeType: string | undefined) => {
+  return Utils.switchCase(
+    attributeType,
+    ['string', () => ''],
+    ['number', () => 0],
+    ['boolean', () => false],
+    ['json', () => ({})]
+  );
+};
+
+export interface WDSAttributeInputProps {
+  autoFocus: boolean;
+  value: any;
+  attributeName: string;
+  dataProvider: object;
+  recordTypeAttributes: { name: string; datatype: string }[];
+  onChange: any;
+  initialValue: any;
+}
+
+const AttributeInput = ({
+  autoFocus = false,
+  value: attributeValue,
+  attributeName,
+  dataProvider,
+  onChange,
+  recordTypeAttributes,
+}: WDSAttributeInputProps) => {
```

Comment:
Nit, but when there are a lot of props, I think a style like this makes the component signature easier to see.

```suggestion
const AttributeInput = (props: WDSAttributeInputProps): ReactNode => {
  const {
    autoFocus = false,
    value: attributeValue,
    attributeName,
    dataProvider,
    onChange,
    recordTypeAttributes,
  } = props;
```